src/core/Xpattern.ml

```diff
@@ -29,7 +29,11 @@ type regexp_string = string [@@deriving show, eq]
 type pattern_id = int [@@deriving show, eq]
 
 type xpattern_kind =
-  | Sem of Pattern.t * Lang.t (* language used for parsing the pattern *)
+  (* opti: parsing Semgrep patterns lazily improves speed significantly.
+   * Parsing 'p/default', which contains more than 1000 rules, including
+   * lots of Ruby rules goes from 13s to just 0.2s (Dyp GLR is pretty slow).
```

Comment:
It's not actually that GLR is slow, it's that it takes a lot of time (0.02 s or so) to create the parser. This work is repeated for each pattern (in `p/default` there were ~600 Ruby patterns), so it adds up. Note that the relevant function is `pp ()` so it isn't doing anything special per pattern.